subprojects/docs/src/docs/userguide/upgrading_version_4.adoc

```diff
@@ -48,24 +49,41 @@ Other notable changes to be aware of that may break your build include:
  * A change that means you should <<#rel4.8:configure_internal_tasks,configure existing `wrapper` and `init` tasks>> rather than defining your own.
  * The <<#rel4.8:pom_wildcard_exclusions,honoring of implicit wildcards in Maven POM exclusions>>, which may result in dependencies being excluded that weren't before.
  * A <<#rel4.6:annotation_processor_configuration,change to the way you add Java annotation processors to a project>>.
+ * The <<#rel5.0:default_memory_settings,default memory settings>> have been reduced.
+ * The <<#rel5.0:default_tool_versions,default versions>> of several code quality plugins have been updated.
+ * <<#rel5.0:library_upgrades,Several libraries>> used by Gradle have been updated.
+ * A change in behavior related to <<#rel5.0:dependency_resolution,dependency resolution>> may impact a small number of users.
+ * There have been several changes to <<#rel5.0:changes_to_default_task,property factory methods on DefaultTask>> that may impact the creation of custom tasks.
 
 [[changes_5.0]]
 == Upgrading from 4.10 and earlier
 
+If you are not already on version 4.10, skip down to the section that applies to your current Gradle version and work your way up until you reach here.
+
+=== Improvements
+
  * The `enableFeaturePreview('IMPROVED_POM_SUPPORT')` flag is no longer necessary
  * The `enableFeaturePreview('STABLE_PUBLISHING')` flag is no longer necessary
+ * Please remove the `--add-modules java.xml.bind` option from `org.gradle.jvmargs`, if set. Gradle now bundles <<#rel5.0:jaxb_and_java9, JAXB>> for Java 9 and above.
 
 === Potential breaking changes
 
 The changes in this section have the potential to break your build, but the vast majority have been deprecated for quite some time and few builds will be affected by a large number of them. We strongly recommend upgrading to Gradle 4.10 first to get a report on what deprecations affect your build.
 
-There is only one change that was not previously deprecated:
+The following breaking changes are not from deprecations, but the result of changes in behavior:
 
  * Optional dependency declarations in POMs no longer influence the version of a dependency that Gradle picks.
 +
 This change is only applicable if you had the `IMPROVED_POM_SUPPORT` preview feature enabled.
  * The evaluation of the `publishing {}` block is no longer deferred until needed but behaves like any other block.
    Please use `afterEvaluate {}` if you need to defer evaluation.
+ * The link:{groovyDslPath}/org.gradle.api.tasks.javadoc.Javadoc.html[`Javadoc`] and link:{groovyDslPath}/org.gradle.api.tasks.javadoc.Groovydoc.html[`Groovydoc`] tasks now delete the destination dir for the documentation before executing. This has been added to remove stale output files from the last task execution.
+ * The <<#java_library_distribution_plugin, Java Library Distribution Plugin>> is now based on the <<#java_library_plugin, Java Library Plugin>> instead of the <<#java_plugin, Java Plugin>>.
```

Comment:
It isn't entirely clear to me what the implications of this change are.  Could we reword this to describe what problems might happen as a result of this change?

Reply:
@gradle/dependency-management ☝️

Reply:
This was added through #5976 so maybe @marcphilipp can comment. But I agree that it looks more like an _improvement_ than a _breaking change_.

Reply:
And I should have read the comments on that PR before answering :wink:. Breaking change it is, though I am not sure how to express it better.

Reply:
Maybe add sth. like the following?

> While it applies the Java Plugin, it behaves slightly different (e.g. it adds the `api` configuration). Thus, make sure to check whether your build behaves as expected after upgrading.

Still kind of vague, though. 😕

Reply:
applied @marcphilipp 's change after this line.